dump/src/reader/v3/mod.rs

```diff
@@ -155,16 +160,43 @@ pub struct V3IndexReader {
 }
 
 impl V3IndexReader {
-    pub fn new(name: String, path: &Path) -> Result<Self> {
+    pub fn new(
+        name: String,
+        path: &Path,
+        index_uuid: &IndexUuid,
+        tasks: BufReader<File>,
+    ) -> Result<Self> {
```

Comment:
```suggestion
    pub fn new(
        index_uuid: &IndexUuid,
        path: &Path,
        tasks: BufReader<File>,
    ) -> Result<Self> {
```

We don’t think the name anymore since it’s already contained in the `index_uuid`